### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by lyskevin**

### CODE REVIEW

File: webproject/src/test/java/com/google/sps/data/MessageTest.java
Comment (lyskevin) [inline webproject/src/test/java/com/google/sps/data/MessageTest.java:35]: The code here is time sensitive so it might not pass 100% of the time.

File: webproject/src/test/java/com/google/sps/authentication/AuthenticationHandlerTest.java
Comment (lyskevin) [inline webproject/src/test/java/com/google/sps/authentication/AuthenticationHandlerTest.java:50]: There is some issue with testing the createLoginURL and createLogoutURL methods in the UserService class (error when making API calls from the test environment) so I excluded them first.

File: webproject/src/test/java/com/google/sps/data/RoomTest.java
Comment (lyskevin) [inline webproject/src/test/java/com/google/sps/data/RoomTest.java:128]: I did not test the firebase methods here (should we do it as part of integration testing instead or create a mock database to test?)

File: webproject/src/test/java/com/google/sps/util/HtmlParserTest.java
Comment (kevinluvian) [inline webproject/src/test/java/com/google/sps/util/HtmlParserTest.java]: remove static here

File: webproject/src/test/java/com/google/sps/util/HtmlParserTest.java
Comment (kevinluvian) [inline webproject/src/test/java/com/google/sps/util/HtmlParserTest.java]: you can put test as return Exception. this way you dont need try catch block on every test.
throwing Exception on test is preferrable since you dont want to have an exception in your test go silently, since the use of test is to catch it

File: webproject/src/test/java/com/google/sps/data/RoomTest.java
Comment (kevinluvian) [inline webproject/src/test/java/com/google/sps/data/RoomTest.java]: is it possible to use List.of or other way here to make it one liner?

File: webproject/src/test/java/com/google/sps/data/RoomTest.java
Comment (kevinluvian) [inline webproject/src/test/java/com/google/sps/data/RoomTest.java]: ditto

Comment (kevinluvian) [reply webproject/src/test/java/com/google/sps/data/RoomTest.java]: also, you should split the content of test function to:
Arrange-Art-Assert

// Arrange
expectedUsers = ...

// Act
room.addUser(user);

// assert
assertEquals..

File: webproject/src/test/java/com/google/sps/data/MessageTest.java
Comment (kevinluvian) [reply webproject/src/test/java/com/google/sps/data/MessageTest.java:35]: when https://github.com/arjavibahety/SpsTeam8/pull/56/files#diff-cc8f223e1684e083dde9ec51abb485e1
 got merged, you should implement a FakeTimestampUtil class to be injected
the FakeTimestampUtil class should be set-able to return a specific date time

File: webproject/src/test/java/com/google/sps/authentication/AuthenticationHandlerTest.java
Comment (kevinluvian) [inline webproject/src/test/java/com/google/sps/authentication/AuthenticationHandlerTest.java]: remove extra line

**Review by kevinluvian — CHANGES_REQUESTED** [review]

File: webproject/src/test/java/com/google/sps/data/RoomTest.java
Comment (lyskevin) [reply webproject/src/test/java/com/google/sps/data/RoomTest.java]: It is not available in Java 8, which we are using for the project. All other list methods or constructors require an existing Collection.

File: webproject/src/test/java/com/google/sps/authentication/AuthenticationHandlerTest.java
Comment (lyskevin) [reply webproject/src/test/java/com/google/sps/authentication/AuthenticationHandlerTest.java]: Done

File: webproject/src/test/java/com/google/sps/data/RoomTest.java
Comment (lyskevin) [reply webproject/src/test/java/com/google/sps/data/RoomTest.java]: Done.

File: webproject/src/test/java/com/google/sps/util/HtmlParserTest.java
Comment (lyskevin) [reply webproject/src/test/java/com/google/sps/util/HtmlParserTest.java]: Done.

File: webproject/src/test/java/com/google/sps/util/HtmlParserTest.java
Comment (lyskevin) [reply webproject/src/test/java/com/google/sps/util/HtmlParserTest.java]: Done.

File: webproject/src/test/java/com/google/sps/data/RoomTest.java
Comment (kevinluvian) [inline webproject/src/test/java/com/google/sps/data/RoomTest.java:112]: add comment on the top of each section to explain that this part is arrange, act, assert
// Arrange
List<String>...

// Act
room.AddUser

// Assert
assertEquals(...)

**Review by kevinluvian — COMMENTED** [review]
LGTM for the other, but lets wait until the DI PR got merged and then make some changes for the timestamp util

File: webproject/src/test/java/com/google/sps/data/RoomTest.java
Comment (lyskevin) [reply webproject/src/test/java/com/google/sps/data/RoomTest.java:112]: Updated

File: webproject/src/test/java/com/google/sps/data/MessageTest.java
Comment (lyskevin) [reply webproject/src/test/java/com/google/sps/data/MessageTest.java:35]: Updated

File: webproject/src/main/java/com/google/sps/servlets/FormHandlerServlet.java
Comment (kevinluvian) [inline webproject/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:74]: create function in firebaesUtil to getMessagesReference

File: webproject/src/main/java/com/google/sps/servlets/FormHandlerServlet.java
Comment (kevinluvian) [inline webproject/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:77]: use a builder

File: webproject/src/main/java/com/google/sps/servlets/TextMessageServlet.java
Comment (kevinluvian) [inline webproject/src/main/java/com/google/sps/servlets/TextMessageServlet.java:71]: ditto

File: webproject/src/test/java/com/google/sps/authentication/AuthenticationHandlerTest.java
Comment (kevinluvian) [inline webproject/src/test/java/com/google/sps/authentication/AuthenticationHandlerTest.java]: split the act and assert

**Review by kevinluvian — CHANGES_REQUESTED** [review]

File: webproject/src/main/java/com/google/sps/servlets/FormHandlerServlet.java
Comment (lyskevin) [reply webproject/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:77]: Arjavi is refactoring this code

File: webproject/src/main/java/com/google/sps/servlets/FormHandlerServlet.java
Comment (lyskevin) [reply webproject/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:74]: Updated

File: webproject/src/main/java/com/google/sps/servlets/TextMessageServlet.java
Comment (lyskevin) [reply webproject/src/main/java/com/google/sps/servlets/TextMessageServlet.java:71]: Same as above

File: webproject/src/test/java/com/google/sps/authentication/AuthenticationHandlerTest.java
Comment (lyskevin) [reply webproject/src/test/java/com/google/sps/authentication/AuthenticationHandlerTest.java]: Updated

**Review by kevinluvian — APPROVED** [review]

**Pull request merged**
